### PR TITLE
More tests

### PR DESCRIPTION
**Pull request by dzmitry-varabei**
Let's make students life a little better! :innocent:

### CODE REVIEW

File: test/max-heap.spec.js [hunk 1]
```diff
@@ -338,6 +338,27 @@ describe('MaxHeap', () => {
...
+          /  \                           /  \
+        15    16   - restoreRoot ->    15   16
+       /  \  /  \                    /  \  /
+     14  13  0   12                14  13  0
+ **/
+
+			const detached = h.detachRoot();
+			h.restoreRootFromLastInsertedNode(detached);
+
+			expect(h.parentNodes.map(n=>n.priority)).to.deep.equal([16,14,13,0]);
```
Comment (lesnitsky) [inline test/max-heap.spec.js:359]: - there is no node with priority `0` pushed to heap
- node with priority 16 is detached, so it shouldn't present in `parentNodes`

File: test/max-heap.spec.js [hunk 2]
```diff
@@ -389,6 +410,34 @@ describe('MaxHeap', () => {
...
+          /  \                           /  \
+        10    20   - shiftDown ->      10   11
+       /  \  /  \                    /  \  /  \
+     5    8 11  6                   5   8 0   6
+ **/
+
+            h.shiftNodeDown(h.root);
+
+            expect(h.root).to.equal(newRoot);
+            expect(h.root.right.left).to.equal(newDeepest);
```
Comment (R1ZZU) [inline test/max-heap.spec.js]: there is no `parentNodes` ([take a look at beforeEach](https://github.com/rolling-scopes-school/priority-queue/pull/1/files#diff-358407ef5166e6d540f40a9eb35cd30eR383))

File: test/max-heap.spec.js (hunk 1 above)
Comment (dzmitry-varabei) [reply test/max-heap.spec.js:359]: 1) Eagle eye! :smile: I will correct all zeros to ones
2) I consider that we use `parentNodes` like this:

``` js
insertNode(node) {
    if (this.root == null){
        this.root = node;
    } else {
        this.parentNodes[0].appendChild(node);
    }
    ///....
}
```

In this case we should have 16 as first element at `parentNodes`. Or you have different implementation?

Comment (lesnitsky) [reply test/max-heap.spec.js:359]: forgot about `beforeEach` block. All good 😏

**Comment by lesnitsky** [comment]
LGTM

**Comment by DaQuirm** [comment]
How did I get into this? 😸 
EDIT: Github kinda answered this question by writing I'm a Rolling Scopes School member on top of this comment 😸 

Reading...

**Pull request merged**